theano/tensor/opt_uncanonicalize.py

```diff
@@ -106,14 +106,14 @@ def local_max_to_min(node):
 def local_alloc_dimshuffle(node):
     if node.op == T.alloc:
         input_ = node.inputs[0]
-        if isinstance(input_.owner.op, DimShuffle):
+        if getattr(input_, 'owner', None) and isinstance(input_.owner.op, DimShuffle):
```

Comment:
variable always have an owner field. So you can simplify the first condition as "input_.owner and ..."